### PR TITLE
Use half duplex in the high-level spec scripts

### PR DESCRIPTION
**Pull request by cmebarrow**
(except where upgrade occurs)
Also added a couple of test cases for abort

### CODE REVIEW

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.gets.response.with.multiple.content.lengths/server.rpt [hunk 1]
```diff
@@ -16,7 +16,6 @@
 
 accept "tcp://localhost:8081"
 accepted
-connected
```
Comment (jfallows) [inline src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.gets.response.with.multiple.content.lengths/server.rpt]: Removed intentionally?

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.should.preserve.unrecognized.headers/server.rpt [hunk 2]
```diff
@@ -21,12 +21,10 @@ accept "http://localhost:8080/"
   option http:transport ${serverTransport}
   option nukleus:route  ${newServerAcceptRef}
   option nukleus:window 8192
-  option nukleus:transmission "duplex"
+  option nukleus:transmission "half-duplex"
 accepted
-connected
```
Comment (jfallows) [inline src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.should.preserve.unrecognized.headers/server.rpt]: Removed intentionally?

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.should.preserve.unrecognized.headers/client.rpt [hunk 3]
```diff
@@ -21,13 +21,11 @@ connect await ROUTED_SERVER
   option http:transport "nukleus://http/streams/source"
   option nukleus:route ${newServerAcceptRef}
   option nukleus:window 8192
-  option nukleus:transmission "duplex"
-connected
```
Comment (jfallows) [inline src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.should.preserve.unrecognized.headers/client.rpt]: Removed intentionally?

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.or.gateway.must.reject.obs.in.header.value/server.rpt [hunk 4]
```diff
@@ -16,7 +16,6 @@
 
 accept "tcp://localhost:8080"
 accepted
-connected
```
Comment (jfallows) [inline src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.or.gateway.must.reject.obs.in.header.value/server.rpt]: Removed intentionally?

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.or.gateway.must.reject.obs.in.header.value/client.rpt [hunk 5]
```diff
@@ -15,7 +15,6 @@
 #
 
 connect "tcp://localhost:8080"
-connected
```
Comment (jfallows) [inline src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.or.gateway.must.reject.obs.in.header.value/client.rpt]: Removed intentionally?

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.gets.response.with.multiple.content.lengths/client.rpt [hunk 6]
```diff
@@ -21,8 +21,7 @@ connect await ROUTED_SERVER
   option http:transport "nukleus://http/streams/source"
   option nukleus:route ${newServerAcceptRef}
   option nukleus:window 8192
-  option nukleus:transmission "duplex"
-connected
```
Comment (jfallows) [inline src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.gets.response.with.multiple.content.lengths/client.rpt]: Removed intentionally?

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/transfer.codings/multiple.requests.transfer.encoding.chunked/server.rpt [hunk 7]
```diff
@@ -32,15 +32,14 @@ read nukleus:begin.ext ${http:header(":path", "/")}
 read nukleus:begin.ext ${http:header(":authority", "localhost:8080")}
 read nukleus:begin.ext ${http:header("content-type", "text/plain")}
 read nukleus:begin.ext ${http:header("transfer-encoding", "chunked")}
+connected
 
 write nukleus:begin.ext ${http:header("status", "200")}
```
Comment (jfallows) [inline src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/transfer.codings/multiple.requests.transfer.encoding.chunked/server.rpt]: Strictly speaking, shouldn't this happen after reading the chunks?

Comment (jfallows) [reply src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/transfer.codings/multiple.requests.transfer.encoding.chunked/server.rpt]: Note: also applies to other similar tests.

File: src/test/java/org/reaktivity/specification/http/rfc7230/ConnectionManagementIT.java [hunk 8]
```diff
@@ -274,6 +274,18 @@ public void concurrentUpgradeRequestsandResponsesWithData() throws Exception
 
     @Test
     @Specification({
+        "${scripts}/request.incomplete.response.headers.and.abort/client",
+        "${scripts}/request.incomplete.response.headers.and.abort/server" })
+    @ScriptProperty("serverTransport \"nukleus://http/streams/source\"")
+    public void requestAndAbortBeforeResponseHeadersComplete() throws Exception
```
Comment (jfallows) [inline src/test/java/org/reaktivity/specification/http/rfc7230/ConnectionManagementIT.java]: Please make sure all IT test names have test names starting with `shouldXXX`.

**Review by jfallows — CHANGES_REQUESTED** [review]

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.gets.response.with.multiple.content.lengths/server.rpt (hunk 1 above)
Comment (cmebarrow) [reply src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.gets.response.with.multiple.content.lengths/server.rpt]: no, fixed

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.should.preserve.unrecognized.headers/server.rpt (hunk 2 above)
Comment (cmebarrow) [reply src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.should.preserve.unrecognized.headers/server.rpt]: no, fixed

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.should.preserve.unrecognized.headers/client.rpt (hunk 3 above)
Comment (cmebarrow) [reply src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.should.preserve.unrecognized.headers/client.rpt]: no, fixed

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.or.gateway.must.reject.obs.in.header.value/server.rpt (hunk 4 above)
Comment (cmebarrow) [reply src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.or.gateway.must.reject.obs.in.header.value/server.rpt]: no, fixed

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.or.gateway.must.reject.obs.in.header.value/client.rpt (hunk 5 above)
Comment (cmebarrow) [reply src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.or.gateway.must.reject.obs.in.header.value/client.rpt]: no fixed

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.gets.response.with.multiple.content.lengths/client.rpt (hunk 6 above)
Comment (cmebarrow) [reply src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/message.format/proxy.gets.response.with.multiple.content.lengths/client.rpt]: no, fixed

File: src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/transfer.codings/multiple.requests.transfer.encoding.chunked/server.rpt (hunk 7 above)
Comment (cmebarrow) [reply src/main/scripts/org/reaktivity/specification/nukleus/http/streams/rfc7230/transfer.codings/multiple.requests.transfer.encoding.chunked/server.rpt]: done

File: src/test/java/org/reaktivity/specification/http/rfc7230/ConnectionManagementIT.java (hunk 8 above)
Comment (cmebarrow) [reply src/test/java/org/reaktivity/specification/http/rfc7230/ConnectionManagementIT.java]: Some method names like clientAndServerMustCloseConnectionAfterRequestWithConnectionClose or serverMustCloseConnectionAfterResponseWithConnectionClose make use of the word "must" to indicate the nature of the requirement in the HTTP specification. Are you okay with names like "mustXxx" in addition to "shouldXxx"? I am assuming not, but that does mean we are losing the "must" information. Also, what do you suggest for names like serverMustCloseConnectionAfterResponseWithConnectionClose? I have gone with shouldCloseConnectionAfterSendingResponseWithConnectionClose.

**Review by jfallows — APPROVED** [review]

**Pull request merged**